### PR TITLE
Feature khwp 34 profile frontend + fixes for demo

### PR DESCRIPTION
**Pull request by Formeme**

### CODE REVIEW

File: frontend/src/components/Authentication/auth.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        }).catch((error) => {
+          if(error.response.data.message){
+            context.$awn.warning(error.response.data.message);
+          }
+
+          this.signout(this);
+          alert(error);
+        })  ;
+  },
+
```
Comment (vlad-shev) [inline frontend/src/components/Authentication/auth.js]: Useful comments?

File: frontend/src/components/Authentication/auth.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            this.user.authenticated = true;
+
+            if (redirect) router.push(redirect);
+        }).catch((error) => {
+          if(error.response.data.message){
+            context.$awn.warning(error.response.data.message);
+          }
+
+          this.signout(this);
+          alert(error);
```
Comment (vlad-shev) [inline frontend/src/components/Authentication/auth.js]: Replace alert with normal notification

File: frontend/src/components/Authentication/auth.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  //       }).catch((error) => {
+  //         alert(error);
+  //       });
+  // },
+
+  signout(context, redirect) {
+    context.$cookies.remove('token');
+    context.$cookies.remove('user_id');
+    this.user.authenticated = false;
+    console.log(context.$cookies);
```
Comment (vlad-shev) [inline frontend/src/components/Authentication/auth.js]: Delete console log

File: frontend/src/components/LoginComponent.vue [hunk 4]
```diff
@@ -28,15 +30,16 @@
...
-          .then((response) => {
-            this.$router.push('/');
-          }).catch((error) => {
-          if (error.response.data.message) {
-            this.$awn.warning(error.response.data.message);
-          }
-          this.userPassword = '';
-        });
+        Auth.authenticate(this, userCredentials, '/');
+      // axios.post('http://localhost:8000/api/users/login/', userCredentials)
```
Comment (vlad-shev) [inline frontend/src/components/LoginComponent.vue]: Comments

File: frontend/src/components/ProfileComponent.vue [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+              this.userAge = response.data['age'];
+              this.userGender = GENDER_CHOISES[response.data['gender']];
+              this.userImage = response.data['profile_image'];
+          }).catch((error) => {
+            Authentication.signout(this);
+            this.$awn.warning(this.error.message);
+          });
+    },
+    save() {
+      console.log(this.userImage);
```
Comment (vlad-shev) [inline frontend/src/components/ProfileComponent.vue]: Delete console log

File: happykh/users/api/views.py [hunk 6]
```diff
@@ -10,9 +10,10 @@
 from rest_framework.permissions import IsAuthenticated, AllowAny
 from rest_framework.authtoken.models import Token
 from rest_framework import exceptions, status
+from smtplib import SMTPException
 from users.models import User
 from users.serializers import LoginSerializer, UserSerializer, PasswordSerializer
-from happykh.settings import EMAIL_HOST_USER
+from happykh.settings import EMAIL_HOST_USER, SENDGRID_API_KEY
```
Comment (vlad-shev) [inline happykh/users/api/views.py]: Delete api key importing

File: frontend/src/components/Authentication/auth.js (hunk 1 above)
Comment (Formeme) [reply frontend/src/components/Authentication/auth.js]: where do you see comments? Did you mean space?

**Pull request merged**